airflow/providers/cncf/kubernetes/utils/pod_manager.py

```diff
@@ -398,14 +417,78 @@ def consume_logs(
                 )
                 time.sleep(1)
 
+    def fetch_requested_container_logs(
```

Comment:
It seems this entire function can be transformed into a generator function instead and use `yield` to get rid of all the `append` calls.

Reply:
Imo, it is simpler to control the flow when we have these append calls. I would like to keep it this way if that is ok by you